server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectSessionService.java

```diff
@@ -382,6 +388,36 @@ public ConnectionSession nullSafeGet(@NotNull String sessionId, boolean autoCrea
         return session;
     }
 
+    public ConnectionSession nullSafeGetSkipPermissionCheck(@NotNull String sessionId,
+            Supplier<ConnectionSession> createConnectionSessionSupplier) {
+        ConnectionSession session = connectionSessionManager.getSession(sessionId);
+        if (session == null) {
+            if (createConnectionSessionSupplier == null) {
+                throw new NotFoundException(ResourceType.ODC_SESSION, "ID", sessionId);
+            }
+            Lock lock = this.sessionId2Lock.computeIfAbsent(sessionId, s -> new ReentrantLock());
```

Comment:
this lock logic can't protect session acquire in same behavior
1. if first attempt get session failed, code goto line 408, but session is created ,will this session should be cancelExpire?
2. each session id will create a lock in sessionId2Lock, this will cause memory leak, is that expected?